src/renderer/tvgGlCanvas.cpp

```diff
@@ -59,8 +59,11 @@ GlCanvas::~GlCanvas()
 }
 
 
-Result GlCanvas::target(int32_t id, uint32_t w, uint32_t h, ColorSpace cs) noexcept
+Result GlCanvas::target(void* context, int32_t id, uint32_t w, uint32_t h, ColorSpace cs) noexcept
 {
+#ifdef __EMSCRIPTEN__
+    emscripten_webgl_make_context_current((EMSCRIPTEN_WEBGL_CONTEXT_HANDLE)context);
```

Comment:
make some check may help improve the performance:
```
if (emscripten_webgl_get_current_context() != context) emscripten_webgl_make_context_current(context);
```

Reply:
gl call should be inside gl_engine, not canvas.